superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -91,16 +89,30 @@ class VisualizeModal extends React.PureComponent {
     });
     if (this.state.chartType === null) {
       hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_CHART_TYPE);
-    } else if (this.state.chartType.requiresTime) {
-      let hasTime = false;
-      for (const colName in cols) {
-        const col = cols[colName];
-        if (col.hasOwnProperty('is_date') && col.is_date) {
-          hasTime = true;
+    } else {
+      if (this.state.chartType.requiresTime) {
+        let hasTime = false;
+        for (const colName in cols) {
+          const col = cols[colName];
+          if (col.hasOwnProperty('is_date') && col.is_date) {
+            hasTime = true;
+          }
+        }
+        if (!hasTime) {
+          hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_TIME);
+        }
+      }
+      if (this.state.chartType.requiresDimension) {
+        const hasMatrix = Object.keys(cols).filter(key => (cols[key].is_dim)).length > 0;
+        if (!hasMatrix) {
+          hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_DIMENSION);
         }
       }
-      if (!hasTime) {
-        hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_TIME);
+      if (this.state.chartType.requiresAggregationFn) {
+        const hasMatrix = Object.keys(cols).filter(key => (cols[key].agg)).length > 0;
+        if (!hasMatrix) {
```

Comment:
could simplify these 2 lines to 1:
`if (Object.keys(cols).filter(key => (cols[key].agg)).length > 0) {`